tests/test_backend_abstract_repr.py

```diff
@@ -65,36 +71,90 @@
 )
 def test_observable_repr(observable, expected_kwargs):
     obs = observable(**expected_kwargs)
-    obs_repr = obs._to_abstract_repr()
-    assert isinstance(obs_repr, dict)
+    obs_repr = json.loads(json.dumps(obs, cls=AbstractReprEncoder))
 
     # test default values
     assert obs_repr["observable"] == obs._base_tag
-    if "evaluation_times" not in expected_kwargs.keys():
-        assert obs_repr["evaluation_times"] is None
-    if "tag_suffix" not in expected_kwargs.keys():
-        assert obs_repr["tag_suffix"] is None
+    assert obs_repr["tag_suffix"] == expected_kwargs.get("tag_suffix", None)
+    if obs_repr["evaluation_times"] is None:
+        assert "evaluation_times" not in expected_kwargs
+    else:
+        assert np.allclose(
+            obs_repr["evaluation_times"], expected_kwargs["evaluation_times"]
+        )
+        assert obs_repr["evaluation_times"] == json.loads(
+            json.dumps(
+                expected_kwargs["evaluation_times"], cls=AbstractReprEncoder
+            )
+        )
 
-    # test kwargs
-    for key, expected_value in expected_kwargs.items():
-        assert obs_repr[key] is expected_value
+    if "one_state" in obs_repr:
+        assert obs_repr["one_state"] == expected_kwargs.get("one_state", None)
+    if "num_shots" in obs_repr:
+        assert obs_repr["num_shots"] == expected_kwargs.get("num_shots", 1000)
```

Comment:
I tend to avoid placing `assert` within `if` blocks because you may accidentaly not ever be reaching them and not know it. In this case, you can again use `get()`
```suggestion
    assert obs_repr.get("one_state", None) == expected_kwargs.get("one_state", None)
    assert obs_repr.get("num_shots", 1000) == expected_kwargs.get("num_shots", 1000)
```